### ISSUE TITLE
Boba Gateway - Metamask Account Switch UI Bug Massive Delay (Rinkeby)

### ISSUE THREAD

**Issue by MitchellUnknown** [omgnetwork/optimism#412]
I will test this issue in the coming days, but here is what i got so far. If you interact with the boba gateway from account 1 on metamask and transfer funds to account 2, it appears that you will encounter UI bugs on the second account. I used fast exits, deposits and other stuff on account 1 and the transaction history page showed all the correct history data. However, when i moved my funds over to account 2 it stopped tracking my wallet history and actions. It was weird, because the wallet worked. I succesfully fast exit some of my funds. But the UI gave me no feedback of the pending fast exits etc.. After 3 to 6 minutes it magically worked again. Very strange. It seems when you have interacted with the boba gateway on one account, it glitches when you want to use it on a second account with in the same metamask root seed. Like i said, need to test this more. Could also be my webbrowser cache that glitched. 

![Boba Gateway - Metamask](https://user-images.githubusercontent.com/90399189/132929117-c0091be2-9a4f-42a5-a461-c69cf8d96997.png)

**Comment by CAPtheorem** [comment]
Expect history and pending data within not more than 5 minutes, although typically it can be much faster - 8 to 12 seconds.

**Comment by CAPtheorem** [comment]
History and pending status is based on the account you are currently set to on MetaMask. If you change to another account, then the history and pending status will show the data only for that specific account. If you have just added that account, and not used it for anything, then the history/pending data sections will be blank. _Notably, there is no support for pending data on inbound transfers_. So for example if you trigger a transfer in one account, and then switch to another account, the `pending` display in account 2 will not show anything. Once the transfer clears to account 2, then your account balances (in that account) will increase. 

It's a great idea to add pending support for the *recipient* account, and also support history of *inbound transfers*, which is also not supported now. But this is not a bug (at least not based on first look) - this is a missing feature.

**Comment by MitchellUnknown** [comment]
> History and pending status is based on the account you are currently set to on MetaMask. If you change to another account (e.g. account 2), then the history and pending status will show the data only for that specific account (account 2). If you have just added that account, and not used it for anything, then the history/pending data sections will be blank. _Notably, there is no support for pending/history data on inbound transfers_.
> 
> For example if you trigger a transfer in one account, and then switch to another account, the `pending` display in account 2 will not show anything. Once the transfer clears to account 2, and is written into the blockchain, and the state roots are pushed to the L1, then your account balances (in account 2) will increase.
> 
> It's a great idea to add pending support for the _recipient_ account, and also support history of _inbound transfers_, which is also not supported now. This is not a bug (whew - at least not based on first look) - but it's a really important feature which will be added as soon as we can.

Yes, i understand that when you send funds from account 1 to account 2 that your history page on the second account will stay empty. As you said, that feature is not yet available. But the bug I encountered was when i moved my funds from account 1 to account 2, and wanted to ‘fast exit’  my funds on account 2, the fast exit worked like intended, but there was no feedback that the fast exit ever took place. Not in the pending transaction box on the wallet page, and not in the transaction history page. Like you said earlier it takes some time. But in general when you start an exit or deposit, and it doesn’t matter on which account 1,2 of 3, it will always instantly scan for blocks in the transaction history page.  So in short, i think when you move your funds internally and try to exit them, the UI glitches at some point. I am not 100% sure, but my guts says there is some kind of error somewhere. But the important thing is, you can exit your funds succesfully. Maybe after 20 sept we could have a deeper look into it. And close the issue afterwards, when we now for sure there is nothing wrong with the UI. Thanks for the respond.
